File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def dictionary(current_word, max_length, alpha, words):
    if len(current_word) > 0:
        words.append(current_word)
    if len(current_word) < max_length:
        for a in alpha:
            dictionary(current_word + a, max_length, alpha, words)

def solution(word):
    answer = 0
    alpha = ['A', 'E', 'I', 'O', 'U']
    words = []
    
    dictionary("", 5, alpha, words)
    
    words.sort()
    
    answer = words.index(word) + 1
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    public int solution(String word) {
        int[] weights = {781, 156, 31, 6, 1};
        Map <Character, Integer> wordDict = new HashMap<>();
        wordDict.put('A', 0);
        wordDict.put('E', 1);
        wordDict.put('I', 2);
        wordDict.put('O', 3);
        wordDict.put('U', 4);
        int answer = 0;
        
        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            answer += wordDict.get(c) * weights[i] + 1;
        }
        
        return answer;
    }
}
